app/ts/components/YearPicker.tsx

```diff
@@ -1,34 +1,39 @@
 import { Signal } from '@preact/signals'
-import * as funtypes from 'funtypes'
 
-interface CreateProps {
-	year: Signal<number>
-}
-
-export const YearPicker = ( { year }: CreateProps) => {
-	const setYear = (text: string) => {
-		const yearToSet = funtypes.Number.parse(text)
-		year.value = yearToSet
-	}
+export const YearPicker = ( { year, validYear }: { year: Signal<number>, validYear: Signal<boolean>  }) => {
 	const increment = () => {
 		year.value = year.value + 1
 	}
+
 	const decrement = () => {
 		year.value = Math.max(year.value - 1, 1)
 	}
-	return <div class = 'year-selector'>
-		<div class = 'year-selector-button year-selector-button-decrement' onClick = { decrement }>&ndash;</div>
-		<div class = 'year-selector-counter'>
-			<input
-				class = 'year-selector-counter-input'
-				maxlength = { 4 }
-				type = 'text'
-				pattern = '\d*'
-				value = { year.value }
-				onInput = { e => setYear(e.currentTarget.value) }
-			/>
-			<div class = 'year-selector-counter-num'>{ year }</div>
+
+	const handleInputChange = (e: Event) => {
+		const inputValue = (e.target as HTMLInputElement).value
+		const parsedYear = parseInt(inputValue, 10)
+		if (!isNaN(parsedYear) && parsedYear > 0) {
+			year.value = parsedYear
+			validYear.value = true
+		} else {
+			validYear.value = false
+		}
+	}
+
+	return (
+		<div class = 'year-selector'>
+			<div class = 'year-selector-button year-selector-button-decrement' onClick = { decrement }>&ndash;</div>
+			<div class = 'year-selector-counter'>
+				<input
+					class = 'year-selector-counter-input'
+					maxLength = { 4 }
+					type = 'text'
+					value = { year.value }
```

Comment:
In general, it is better to provide `Signal` objects rather than the `.value` on them to components.  This minimizes re-renders and is generally a good habit to follow.
```suggestion
					value = { year }
```